draft-ietf-taps-arch.md

```diff
@@ -115,6 +115,82 @@ The goal of the Transport Services architecture is to provide a common, flexible
 
 This document is developed in parallel with the specification of the Transport Services API {{I-D.ietf-taps-interface}} and Implementation {{I-D.ietf-taps-impl}} documents.
 
+## Overview
+
+The model for using sockets for networking can be represented as follows: applications create connections and transfer data using the socket API, which provides the interface to the implementations of UDP and TCP (typically implemented in the system's kernel), which in turn send data over the available network layer interfaces.
+
+~~~~~~~~~~
+
++-----------------------------------------------------+
+|                    Application                      |
++-----------------------------------------------------+
+                          |
++-----------------------------------------------------+
+|                    Socket API                       |
++-----------------------------------------------------+
+                          |
++-----------------------------------------------------+
+|            Kernel Protocol Implementation           |
+|                    (UDP, TCP)                       |
++-----------------------------------------------------+
+                          |
++-----------------------------------------------------+
+|               Network Layer Interface               |
++-----------------------------------------------------+
+
```

Comment:
I like this, but would propose to update it as follows:

```
 +-----------------------------------------------------+
 |                    Application                      |
 +-----------------------------------------------------+
        |                  |                    |
 +-----------------------------------------------------+
 |      |             Socket API                |      |
 +-----------------------------------------------------+
        |                                       |
 +-----------------------------------------------------+
 |      |     Kernel Protocol Implementation    |      |
 |     TCP                                     UDP     |
 +-----------------------------------------------------+
                           |
 +-----------------------------------------------------+
 |               Network Layer Interface               |
 +-----------------------------------------------------+
```

Reply:
What are the lines from the application down to TCP and UDP? Just showing the different use cases?

Reply:
No, the point was to show that applications are statically bound to either TCP or UDP.

Reply:
Ah, I see now. That makes more sense now. I'll try to redraw to illustrate that.

Reply:
Okay, take a look at the updated diagram

Reply:
Much nicer! Thanks for this!